src/rewrite.rs

```diff
@@ -137,7 +137,8 @@ impl Wizer {
                 s if s.id == SectionId::Export.into() => {
                     let mut exports = wasm_encoder::ExportSection::new();
                     for export in module.exports(cx) {
-                        if export.field == self.init_func
+                        if (!self.keep_init_func.unwrap_or(DEFAULT_KEEP_INIT_FUNC)
+                            && export.field == self.init_func)
```

Comment:
There is a similar bit of code for when the Wasm uses module linking that also needs to be updated in the same way:

https://github.com/bytecodealliance/wizer/blob/5dbfd4eade422acdff4c019941addf2656213df7/src/rewrite.rs#L424